ibis/expr/operations/udf.py

```diff
@@ -19,12 +21,24 @@
 from ibis.common.deferred import deferrable
 
 if TYPE_CHECKING:
+    from collections.abc import Iterable, MutableMapping
+
     import ibis.expr.types as ir
 
 
 EMPTY = inspect.Parameter.empty
 
 
+_udf_name_cache: MutableMapping[
+    type[ops.Node], Iterable[int]
+] = collections.defaultdict(itertools.count)
+
+
+def _make_udf_name(name: str) -> str:
```

Comment:
Is this for generating unique UDF names?

Reply:
Yes, it already existed in the bigquery backend.